### PR TITLE
DTSSB-2111  (Link in checkboxes label)

### PR DESCRIPTION
**Pull request by planctus**
The reason for the latest changes is that we didn't consider the case of a "required" boolean field.
In that case the label is also containing the span holding the * character. 

The related test is available in the devkit:
https://github.com/ec-europa/ec-europa-theme-devkit
file name is: link_in_checkboxes.feature

This is a build where this test has run using the DTTSB-2111 branch in the europa theme.
https://continuousphp.com/git-hub/ec-europa/ec-europa-theme-devkit/build/93dc996e-db80-4ed6-9712-608f62721941/output

### CODE REVIEW

**Comment by planctus** [comment]
The reason for the latest changes is that we didn't consider the case of a "required" boolean field.
In that case the label is also containing the span holding the * character. 

The related test is available in the devkit:
https://github.com/ec-europa/ec-europa-theme-devkit
file name is: link_in_checkboxes.feature

This is a build where this test has run using the DTTSB-2111 branch in the europa theme.
https://continuousphp.com/git-hub/ec-europa/ec-europa-theme-devkit/build/93dc996e-db80-4ed6-9712-608f62721941/output

File: template.php [hunk 1]
```diff
@@ -275,6 +288,17 @@ function europa_form_element(&$variables) {
 
       $output .= $feedback_message;
   }
+  // @todo This would be better handled in a theme_form_element_label override.
```
Comment (nelsongrin) [inline template.php]: Please create a ticket and add it to the comments.

File: template.php [hunk 2]
```diff
@@ -275,6 +288,17 @@ function europa_form_element(&$variables) {
...
+  if (!empty($anchor)) {
+    // Add the link.
+    $output .= $anchor;
+
+    if (!empty($variables['element']['#required'])) {
+      // Find the form-required span and put it after the link.
+      preg_match('/<span class="form-required\b[^>]*>(.*?)<\/span>/is', $output, $required);
```
Comment (nelsongrin) [inline template.php]: Here I think you could use the preg_replace to avoid doing the preg_match and str_replace.

**Comment by nelsongrin** [comment]
Tested and code works as expected.
Behat test approved.

**Pull request merged**